### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by jajik**
A few things to consider

- does it make sense to have matrices private in joints (like `RotationalJoint`) and getters for them? Maybe it would be better to have them public and `const`...
- how about targets for CMake? Now we have `configuration` and `test-configuration` and the second one is linked with a json library. We do not want to have json as a dependecy of `configuration` itself, however, it is necessary when someone wants to use the serialization. Therefore,  does it make sense to have `configuration` and `configuration-with-json` when someone wants to use it, or is it user's responsibility to add it?

### CODE REVIEW

**Comment by Ondraceq** [comment]
I think that having the matrix private and have getters is better.

I would have a target `configurationJson` that links the `configuration` as well (since it depends on it).
Also note the camelCase in the target - we only use `rofi-` and `test-` right now as prefixes, but otherwise it should be camelCase for consistency.

File: softwareComponents/configuration/include/configuration/joints.hpp [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 namespace rofi::configuration {
 
+using namespace rofi::configuration::matrices;
```
Comment (Ondraceq) [inline softwareComponents/configuration/include/configuration/joints.hpp:17]: This is polluting namespace `rofi::configuration`. If you want to have it in this namespace, remove the `matrices` namespace.

I think that using it in a `cpp` file is fine, but not in a header file.

**Comment by jajik** [comment]
* check needed for `isValid`, its constness and calling of `prepare`

File: softwareComponents/configuration/include/configuration/joints.hpp (hunk 1 above)
Comment (jajik) [reply softwareComponents/configuration/include/configuration/joints.hpp]: Should be fixed by now. Now it uses only the two types; `Matrix` and `Vector`.

**Review by yaqwsx — APPROVED** [review]

**Pull request merged**